packages/vats/src/localchain.js

```diff
@@ -67,20 +124,23 @@ const prepareLocalChainAccount = zone =>
        * @param {Payment<'nat'>} payment
        * @param {Pattern} [optAmountShape] throws if the Amount of the Payment
        *   does not match the provided Pattern
-       * @returns {Promise<Amount>}
+       * @returns {PromiseVow<Amount<'nat'>>}
        */
       async deposit(payment, optAmountShape) {
-        const { bank } = this.state;
+        const { bank, depositPaymentHandler } = this.state;
 
-        const allegedBrand = await E(payment).getAllegedBrand();
-        const purse = E(bank).getPurse(allegedBrand);
-        return E(purse).deposit(payment, optAmountShape);
+        const getBrandWatcher = watch(E(payment).getAllegedBrand());
+        return watch(getBrandWatcher, depositPaymentHandler, {
```

Comment:
Why the double watch here?
```suggestion
        return watch(E(payment).getAllegedBrand(), depositPaymentHandler, {
```

Reply:
Still learning conventions and thought this was more readable. Aside - it would be awesome to have a style guide / best practices doc for these (naming is tough as well). Happy to apply these changes

Reply:
The result of an eventual send is durably watchable, so there is no need to wrap it in a vow, which is what `watch` without a handler effectively does.

You can keep it split over 2 lines, without the first `watch`, in which case I'd just name it result, or whatever the convention is (I'm am not the best source for that)

```suggestion
        const getBrandResult = E(payment).getAllegedBrand();
        return watch(getBrandResult, depositPaymentHandler, {
```